src/coreclr/jit/morph.cpp

```diff
@@ -9820,95 +9820,50 @@ GenTree* Compiler::fgMorphForRegisterFP(GenTree* tree)
 //      ignoreUsedInSIMDIntrinsic - bool. If this is set to true, then this function will ignore
 //                                  the UsedInSIMDIntrinsic check.
 //
-// return value:
+// Return Value:
 //       A GenTree* which points the simd lclvar tree belongs to. If the tree is not the simd
 //       instrinic related field, return nullptr.
 //
-
 GenTree* Compiler::getSIMDStructFromField(GenTree*     tree,
                                           CorInfoType* simdBaseJitTypeOut,
                                           unsigned*    indexOut,
                                           unsigned*    simdSizeOut,
                                           bool         ignoreUsedInSIMDIntrinsic /*false*/)
 {
-    GenTree* ret = nullptr;
-    if (tree->OperGet() == GT_FIELD)
+    if (tree->OperIs(GT_FIELD))
     {
         GenTree* objRef = tree->AsField()->GetFldObj();
-        if (objRef != nullptr)
+        if ((objRef != nullptr) && objRef->OperIs(GT_ADDR))
         {
-            GenTree* obj = nullptr;
-            if (objRef->gtOper == GT_ADDR)
-            {
-                obj = objRef->AsOp()->gtOp1;
-            }
-            else if (ignoreUsedInSIMDIntrinsic)
-            {
-                obj = objRef;
-            }
```

Comment:
We don't need this anymore because now that GT_CALL gets the handle directly from the arg info, we should never encounter a `OBJ(SIMD)` or `OBJ(HWINTRINSIC)`, right?

Reply:
This condition specifically used to account for the eager morphing of implicit byrefs, that is, we used to get here with an already morphed implicit byref: `FIELD(LCL_VAR byref)`. After #70243, we no longer do, the tree is always `FIELD(ADDR(LCL_VAR struct))`.

For the below code which looks at `ADDR(HWI/SIMD)`, yep, it's the result of us no longer generating such IR.